### PR TITLE
Support sysadvisor metrics emitting

### PR DESCRIPTION
**Pull request by waynepeking348**
#### What type of PR is this?
Features

#### What this PR does / why we need it:
1. add more metrics to support more complex strategies in sysadvisor
2. export more variables in sysadvisor region

### CODE REVIEW

File: pkg/util/metric/store_util.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+// AggregatePodNumaMetric handles numa-level metric for all pods
+func (c *MetricStore) AggregatePodNumaMetric(podList []*v1.Pod, numa, metricName string, agg Aggregator) float64 {
+	sumMetric := 0.
+	validPods := sets.NewString()
+	for _, pod := range podList {
+		if validPods.Has(string(pod.UID)) {
+			continue
+		}
+
+		for _, container := range pod.Spec.Containers {
```
Comment (sun-yuliang) [inline pkg/util/metric/store_util.go:43]: Do we need to include non-main container metrics or main container only?

File: pkg/agent/sysadvisor/types/types.go [hunk 2]
```diff
@@ -112,8 +113,16 @@ type PoolInfo struct {
 
 // RegionInfo contains region information generated by sysadvisor resource advisor
 type RegionInfo struct {
-	ControlKnobMap ControlKnob
-	RegionType     QoSRegionType
+	RegionType  QoSRegionType
```
Comment (sun-yuliang) [inline pkg/agent/sysadvisor/types/types.go:116]: BindingNuma -> BindingNumas
ProvisionTopPriority -> ProvisionPolicyTopPriority

File: pkg/agent/sysadvisor/plugin/qosaware/resource/cpu/region/region_base.go [hunk 3]
```diff
@@ -66,21 +66,24 @@ type QoSRegionBase struct {
...
-	// headroomPolicies for comparing and merging different headroom policy results, the former has higher priority
-	headroomPolicies []*internalHeadroomPolicy
+	// provisionPolicies for comparing and merging different provision policy results,
+	// the former has higher priority; provisionPolicyInUse indicates the provision policy
+	// that is in-use currently
+	provisionPolicies    []*internalProvisionPolicy
+	provisionPolicyInUse *internalProvisionPolicy
+	// headroomPolicies for comparing and merging different headroom policy results,
+	// the former has higher priority; headroomPolicyInUse indicates the provision policy
+	//	// that is in-use currently
```
Comment (sun-yuliang) [inline pkg/agent/sysadvisor/plugin/qosaware/resource/cpu/region/region_base.go:84]: typo

Comment (waynepeking348) [reply pkg/agent/sysadvisor/plugin/qosaware/resource/cpu/region/region_base.go]: > typo

fixed

File: pkg/agent/sysadvisor/plugin/qosaware/resource/cpu/advisor.go [hunk 4]
```diff
@@ -174,30 +174,30 @@ func (cra *cpuResourceAdvisor) update() {
 	cra.mutex.Lock()
 	defer cra.mutex.Unlock()
 
-	// Check if essential pool info exists. Skip update if not in which case sysadvisor
+	// check if essential pool info exists. skip update if not in which case sysadvisor
```
[the remark this answers is not in the document]
Comment (waynepeking348) [reply pkg/agent/sysadvisor/plugin/qosaware/resource/cpu/advisor.go:177]: > Why not capitalized?

no sepecific reasons, just want to align with the comment formats with others (also for personal coding styles)

**Review by sun-yuliang — APPROVED** [review]

**Review by luomingmeng — APPROVED** [review]

**Pull request merged**